### DISCUSSION TITLE
Query regarding AVB simulation

### DISCUSSION THREAD

**Discussion by debnathr** [CoRE-RG/CoRE4INET#34] [Q&A] [answered]
Hello,

I have a query regarding the frame interval of stream A and B. The interval of stream A and B are as follows. 

//Service class A interval is 125us
#define SR_CLASS_A_INTERVAL "0.000125s"

//Service class B interval is 250us
#define SR_CLASS_B_INTERVAL "0.000250s"
 
Now if I want to send class A stream with a period of 2000us how do I configure it? If in the node1.ini file, the "intervalFrames" default value is 1 this means 1 class-A frame will be sent in 125us. So, how do I configure it so that the node1 sends 1 class-A frame in 2000us instead of 125us.

It will be helpful if you could clarify this.

Thank you.

**Comment by Musteblume** [comment, marked as answer]
Hi,

there is a AVBCustomIntervalSourceApp where you can specify a custom interval.

in the ini type:

```
**.<nodename>.app[<number>].typename = "AVBCustomIntervalSourceApp"
... <Default AVB Source Configs>
**.<nodename>.app[<number>].interval = 2000us
```

Hope this Helps!

Regards
Philipp

**Comment by debnathr** [comment]
Thank you so much. It helped.
